numba/tests/test_fancy_indexing.py

```diff
@@ -3,206 +3,163 @@
-from numba import jit, typeof
+from numba import njit, typeof
 from numba.core import types
 from numba.core.errors import TypingError
 from numba.tests.support import MemoryLeakMixin, TestCase, tag
 
 
-def getitem_usecase(a, b):
-    return a[b]
-
-def setitem_usecase(a, idx, b):
-    a[idx] = b
-
-def np_take(A, indices):
-    return np.take(A, indices)
-
-def np_take_kws(A, indices, axis):
-    return np.take(A, indices, axis=axis)
-
 class TestFancyIndexing(MemoryLeakMixin, TestCase):
-
-    def generate_advanced_indices(self, N, many=True):
-        choices = [np.int16([0, N - 1, -2])]
-        if many:
-            choices += [np.uint16([0, 1, N - 1]),
-                        np.bool_([0, 1, 1, 0])]
-        return choices
-
-    def generate_basic_index_tuples(self, N, maxdim, many=True):
-        """
-        Generate basic index tuples with 0 to *maxdim* items.
-        """
-        # Note integers can be considered advanced indices in certain
-        # cases, so we avoid them here.
-        # See "Combining advanced and basic indexing"
-        # in http://docs.scipy.org/doc/numpy/reference/arrays.indexing.html
-        if many:
-            choices = [slice(None, None, None),
-                       slice(1, N - 1, None),
-                       slice(0, None, 2),
-                       slice(N - 1, None, -2),
-                       slice(-N + 1, -1, None),
-                       slice(-1, -N, -2),
-                       ]
-        else:
-            choices = [slice(0, N - 1, None),
-                       slice(-1, -N, -2)]
-        for ndim in range(maxdim + 1):
-            for tup in itertools.product(choices, repeat=ndim):
-                yield tup
-
-    def generate_advanced_index_tuples(self, N, maxdim, many=True):
-        """
-        Generate advanced index tuples by generating basic index tuples
-        and adding a single advanced index item.
-        """
-        # (Note Numba doesn't support advanced indices with more than
-        #  one advanced index array at the moment)
-        choices = list(self.generate_advanced_indices(N, many=many))
-        for i in range(maxdim + 1):
-            for tup in self.generate_basic_index_tuples(N, maxdim - 1, many):
-                for adv in choices:
-                    yield tup[:i] + (adv,) + tup[i:]
-
-    def generate_advanced_index_tuples_with_ellipsis(self, N, maxdim, many=True):
-        """
-        Same as generate_advanced_index_tuples(), but also insert an
-        ellipsis at various points.
-        """
-        for tup in self.generate_advanced_index_tuples(N, maxdim, many):
-            for i in range(len(tup) + 1):
-                yield tup[:i] + (Ellipsis,) + tup[i:]
-
-    def check_getitem_indices(self, arr, indices):
-        pyfunc = getitem_usecase
-        cfunc = jit(nopython=True)(pyfunc)
+    # (shape or array, indices)
+    # Every case has exactly one, one-dimensional array,
+    # Otherwise it's not fancy indexing
+    indexing_cases = [
```

Comment:
I'm not sure that this list of cases covers the same range of cases as previously presented with the generated inputs. I gather this patch is mostly about better enumeration of the failure modes in typing, however, I'm concerned that removing the existing (what appears to be more broad) range of test inputs means that lowering/runtime checks are not being exercised so thoroughly.